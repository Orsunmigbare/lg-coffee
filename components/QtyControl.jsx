import React, { useState } from 'react';
import Alert from './Alert';

const QtyControl = (props) => {
  const [count, setCount] = useState(1);
  const [alert, setAlert] = useState(null);

  const handleCount = (operator) => {
    if (count === 0) return;
    if (operator === '+') setCount(count + 1);
    else if (operator === '-') setCount(count - 1);
    else {
      setAlert({
        status: 'danger',
        message: 'Unknown error occured. Try again later.',
      });
    }
  };

  const handleSubmit = () => {
    if (count === 0)
      setAlert({ status: 'warning', message: 'Nothing to add!' });
    else {
      setAlert({
        status: 'success',
        message: 'Successfully added products to cart!',
      });
      setTimeout(() => {
        setAlert(null);
      }, 3000);
    }
  };

  return (
    <div className="d-flex flex-column align-items-end">
      <div className="btn-group">
        <button
          type="button"
          className="btn btn-outline-dark p-1 pr-3 pl-3"
          style={{ height: 'fit-content' }}
          onClick={() => handleCount('-')}
        >
          -
        </button>
        <p className="d-flex pr-3 pl-3" style={{ height: 'fit-content' }}>
          {count}
        </p>
        <button
          type="button"
          className="btn btn-outline-dark p-1 pr-3 pl-3"
          style={{ height: 'fit-content' }}
          onClick={() => handleCount('+')}
        >
          +
        </button>
      </div>
      <button
        type="button"
        className="btn btn-outline-dark qty_submit"
        onClick={handleSubmit}
      >
        Add To Cart
      </button>
      <div style={{ height: '100px' }}>
        {alert ? <Alert status={alert} /> : null}
      </div>
    </div>
  );
};

export default QtyControl;
