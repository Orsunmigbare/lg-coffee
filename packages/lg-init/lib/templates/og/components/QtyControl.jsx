/* eslint-disable no-console */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Form from 'react-bootstrap/Form';
import Alert from './Alert';
import { H5 } from './_helpers';

const QtyControl = ({ product, cart, handleUpdatedCartInState }) => {  
  const [alert, setAlert] = useState(null);

  const handleSubmit = (ev) => {
    ev.preventDefault();

    let updatedCart;

    if (cart) {
      updatedCart = JSON.stringify({
        ...cart,
        [product.sku]: {
          ...product,
          qty: ev.target.qty.value,
        },
      });
    } else {
      // if cart empty, set cart
      updatedCart = JSON.stringify({
        [product.sku]: {
          ...product,
          qty: ev.target.qty.value,
        },
      });
    }

    localStorage.setItem('cart', updatedCart);
    handleUpdatedCartInState(updatedCart)
    setAlert({
      status: 'success',
      message:
        ev.target.qty.value === 1
          ? `added ${ev.target.qty.value} product to your cart!`
          : `added ${ev.target.qty.value} products to your cart!`,
    });
    setTimeout(() => {
      setAlert(null);
    }, 3000);
  
  };

  return (
    <div className="d-flex justify-content-end">
      <div
        style={{
          height: '100px',
          position: 'absolute',
          zIndex: '2000',
          left: '1em',
          bottom: '-1em',
        }}
      >
        {alert ? <Alert status={alert.status}>{alert.message}</Alert> : null}
      </div>

      <div className="d-flex flex-column align-items-end mb-3">
        <Form className="d-flex align-items-center" onSubmit={handleSubmit}>
          <H5 className="p-0 m-0">QTY:</H5>  
          <Form.Control 
            type="number" 
            label="qty"
            min="1" 
            max="40" 
            name="qty"
            defaultValue="1"
            className="border-dark mx-3"
            style={{
              width: 'fit-content'
            }}
          /> 
          <button
            type="submit"
            className="btn btn-outline-dark qty_submit"
            style={{width: '125px !important'}}
          >
            Add To Cart
          </button>
        </Form>
      </div>
    </div>
  );
};

QtyControl.propTypes = {
  product: PropTypes.shape({
    sku: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
  }).isRequired,
  cart: PropTypes.shape({}),
  handleUpdatedCartInState: PropTypes.func.isRequired,
};


QtyControl.defaultProps = {
  cart: {},
}

export default QtyControl;
