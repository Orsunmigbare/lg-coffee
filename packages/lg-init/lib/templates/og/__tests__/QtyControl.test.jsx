import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import QtyControl from '../components/QtyControl';

const sku = 'demo_sku';

const product = {
  name: 'product',
  description: 'this is an awesome product you should buy 2',
  price: 287,
  sku: 'abc_123',
  images: [
    '/path_01.png',
    '/path_02.png',
    '/path_03.png',
  ],
  qty: 12,
};

it('QtyControl component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<QtyControl sku={sku} product={product} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('QtyControl component renders unchanged', () => {
  const tree = renderer.create(<QtyControl sku={sku} product={product} />).toJSON();
  expect(tree).toMatchSnapshot();
});
