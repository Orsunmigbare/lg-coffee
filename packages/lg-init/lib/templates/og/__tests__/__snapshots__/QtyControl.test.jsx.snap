// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QtyControl component renders unchanged 1`] = `
<div
  className="d-flex flex-column align-items-end"
>
  <div
    className="btn-group"
  >
    <button
      className="btn btn-outline-dark p-1 pr-3 pl-3"
      onClick={[Function]}
      style={
        Object {
          "height": "fit-content",
        }
      }
      type="button"
    >
      -
    </button>
    <p
      className="d-flex pr-3 pl-3"
      style={
        Object {
          "height": "fit-content",
        }
      }
    >
      1
    </p>
    <button
      className="btn btn-outline-dark p-1 pr-3 pl-3"
      onClick={[Function]}
      style={
        Object {
          "height": "fit-content",
        }
      }
      type="button"
    >
      +
    </button>
  </div>
  <button
    className="btn btn-outline-dark qty_submit"
    onClick={[Function]}
    type="button"
  >
    Add To Cart
  </button>
  <div
    style={
      Object {
        "height": "100px",
      }
    }
  />
</div>
`;
